src/librbd/librbd.cc

```diff
@@ -2963,6 +2963,7 @@ extern "C" int rbd_list2(rados_ioctx_t p, rbd_image_spec_t *images,
   TracepointProvider::initialize<tracepoint_traits>(get_cct(io_ctx));
   tracepoint(librbd, list_enter, io_ctx.get_pool_name().c_str(),
              io_ctx.get_id());
+  memset(images, 0, sizeof(*images) * *size);
```

Comment:
Technically these shouldn't be needed (except on the error return code path) since the cleanup functions take a count, but I suppose it's just cleaner to do it once.